lib/stack_master/aws_driver/cloud_formation.rb

```diff
@@ -33,9 +44,16 @@ def set_region(value)
       private
 
       def cf
-        @cf ||= Aws::CloudFormation::Client.new(region: region, retry_limit: 10)
-      end
+        @cf ||= begin
+          params = {
+            region: region,
+            retry_limit: 10,
+          }
+          params[:credentials] = Aws::SharedCredentials.new(profile_name: profile_name) if profile_name
```

Comment:
It seems like it would be easier to just over-write the environment variable rather than pass it down the stack.

Reply:
From memory I did it this way so that if you simply do ```stack_master apply``` different stacks can have different profiles. i.e. the Chine situation